src/shared/components/modal/ModalProvider.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { ModalPortal } from '../../../ModalPortal';
+
+type ModalContextType = {
+  props?: React.ReactElement;
+  setProps?: (children: React.ReactElement) => void;
+};
+
+export const ModalContext = React.createContext<ModalContextType>({});
+export const ModalProvider: React.FC<ModalContextType> = ({ children }) => {
+  const [context, setContext] = React.useState<React.ReactElement>();
+
+  React.useEffect(() => {
+    setContext(null);
+  }, []);
+
+  return (
+    <>
+      <ModalPortal />
+      <ModalContext.Provider value={{ props: context, setProps: setContext }}>
```

Comment:
this is misleading, you are naming context => props. And you don't need to pass context to the provider. the provider just needs to have a callback that can open the modal